src/ray/raylet/node_manager.cc

```diff
@@ -542,6 +548,53 @@ ray::Status NodeManager::RegisterGcs() {
         "NodeManager.deadline_timer.print_event_loop_stats");
   }
 
+  if (RayConfig::instance().use_ray_syncer()) {
```

Comment:
should we create a helper function for this?

Reply:
Yeah, I agree this could be better. but given the current pattern here, we just put all init-related things inside this function, and this is not very complicated, it should be OK I guess? What do you think?